src/commands/dev/index.ts

```diff
@@ -409,6 +412,52 @@ export default class Dev extends BaseCommand {
     return parsed;
   }
 
+  /**
+   * Cleanup architect-created images and layers that are no longer in use to prevent
+   * old images from wasting space on disk.
+   */
+  async pruneImages(compose: DockerComposeTemplate): Promise<void> {
+    // Remove build cache layers with our label that are older than 7 days
+    await docker(['builder', 'prune', '--filter', `label=${DOCKER_IMAGE_LABEL}`, '--filter', `unused-for=${7 * 24}h`, '-af'], { stdout: false });
+    // Remove images that are untagged and created by previous `architect dev` runs
+    await docker(['image', 'prune', '--filter', `label=${DOCKER_IMAGE_LABEL}`, `--filter`, 'dangling=true', '-f'], { stdout: false });
+
+    // Find the set of services and name of all components being deployed. Any image that is labelled with the name
+    // of a component being run but uses a Repository name that no longer matches any service names can be
+    // removed. This can happen when changing the name of a service in the architect.yml.
```

Comment:
Is this a fair assumption? Someone might change their mind and rename their component to something they used five minutes ago. I don't love not putting some kind of time restriction on these operations

Reply:
Hmm that's fair, I think in most cases of quick name waffling the docker layer cache will do the heavy lifting and the user wouldn't even know we cleared the previously used image name (e.g. if you just change image name and none of the layers, nothing really gets rebuilt except metadata about the image). Interested in more opinions here (if you have a sec @mueschm) 

One of the downsides to time gated cache clearing is if you're doing something that's causing old images to constantly get created and become danging (like, changing the name of a service + making changes that trigger large rebuilds) you'll rapidly fill your disk with garbage and waiting 7d to clean it would leave users with issues still.